spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Linear.scala

```diff
@@ -35,6 +37,13 @@ class Linear(
   private val initGradWeight: Tensor[Float] = null,
   private val initGradBias: Tensor[Float] = null) extends MklDnnLayer with Initializable {
 
+  object Extend extends Serializable {
```

Comment:
Looks like other layers have similar logic. Can we doing some code refactor here?